zaza/utilities/bundle.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for disfiguring Juju bundles."""
+
+
+import argparse
+import yaml
+import sys
+
+
+def remove_machine_specification(input_yaml):
+    """
+    Will remove the machien specifications from a supplied bundle.
+
+    :param input_yaml: Juju bundle to strip machinens from
+    :type input_yaml: dict
+    :returns: bundle without machine placements
+    :rtype: dict
+    """
+    machines = input_yaml.pop("machines", None)
+    if not input_yaml.get('series'):
+        if machines:
+            for (name, details) in machines.items():
+                if details['series']:
+                    input_yaml['series'] = details['series']
```

Comment:
What if there is a mix of series ?

Reply:
Then we don't support that ;-) Per some discussion with @andrewdmcleod, it could also be interesting to be able to override the series in here, but that's an open work item

Reply:
ok, thats fine, but detect it if it happens  and explode.